File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.dart
class Solution {
  int maxProfit(List<int> prices) {
    int l = 0, r=0;
    int maxP = 0;

    while(r < prices.length){

      if(prices[r]<prices[l]){
        l = r;
      }

      int curP = prices[r] - prices[l];
      maxP = curP > maxP ? curP : maxP;  // maxP = max(maxP, curP)  import 'dart:math';



      r++;
    }


    return maxP;
  }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minSoFar = float('inf')
        profit = 0
        for p in prices:
            minSoFar = min(minSoFar, p)
            profit = max(profit, (p-minSoFar))

        return profit
        